oscarapi/serializers/checkout.py

```diff
@@ -195,11 +196,21 @@ def get_initial_order_status(self, basket):
 
     def validate(self, attrs):
         request = self.context['request']
- 
+
         if request.user.is_anonymous() and not settings.OSCAR_ALLOW_ANON_CHECKOUT:
             message = _('Anonymous checkout forbidden')
             raise serializers.ValidationError(message)
 
+        if request.user.is_anonymous():
+            if 'guest_email' not in attrs:
+                # Always require the guest email field if the user is anonymous
+                message = _('Guest email is required for anonymous checkouts')
+                raise serializers.ValidationError(message)
+        else:
+            if 'guest_email' in attrs:
+                # Don't store guest_email field if the user is authenticated
+                attrs['guest_email'] = None
+
```

Comment:
I would say to write it like this:
```python
        if request.user.is_anonymous():
            if not settings.OSCAR_ALLOW_ANON_CHECKOUT:
                message = _('Anonymous checkout forbidden')
                raise serializers.ValidationError(message)

            if 'guest_email' not in attrs:
                # always require the guest email field if the user is anonymous
                message = _('Guest email is required for anonymous checkouts')
                raise serializers.ValidationError(message)
        else:
            # delete guest_email when it's there because we are authenticated
            if 'guest_email' in attrs:
                del attrs['guest_email']
```